File: timemanager-service/src/main/java/com/time/timemanager/authentication/AuthService.java
package com.time.timemanager.authentication;

import com.time.timemanager.authentication.dtos.*;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;

public interface AuthService {
    ResponseEntity<?> register(final RegisterRequest request);

    ResponseEntity<?> confirmAccount(final String token);

    ResponseEntity<?> login(final LoginRequest request);

    ResponseEntity<?> refreshToken(final String refreshToken);

    ResponseEntity<?> logout();

    ResponseEntity<?> requestPasswordReset(final PasswordResetInitRequest request);

    ResponseEntity<?> resetPassword(final PasswordResetSubmitRequest request);

    ResponseEntity<?> deleteUser(final Authentication auth);

    ResponseEntity<?> reactivateUser(final ReactivateUserRequest request);

    ResponseEntity<?> getUserStatus(final String email);

    ResponseEntity<?> getUserDetails(final Authentication auth);
}

File: timemanager-app/src/app/auth/auth.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  private apiUrl = '/api/auth';

  constructor(private http: HttpClient) {}

  login(data: { email: string, password: string }): Observable<any> {
    return this.http.post(`${this.apiUrl}/login`, data);
  }

  register(data: { username: string; email: string; password: string }): Observable<any> {
    return this.http.post(`${this.apiUrl}/register`, data);
  }

  confirmAccount(token: string): Observable<any> {
    return this.http.post(`${this.apiUrl}/confirm?token=${encodeURIComponent(token)}`, null);
  }

  refreshToken(): Observable<any> {
    return this.http.post(`${this.apiUrl}/refresh`, {}, { withCredentials: true });
  }

  logout(): Observable<any> {
    return this.http.post(`${this.apiUrl}/logout`, {}, { withCredentials: true });
  }

  requestPasswordReset(data: { email: string }): Observable<any> {
    return this.http.post(`${this.apiUrl}/reset-request`, data);
  }

  resetPassword(data: { token: string, newPassword: string }): Observable<any> {
    return this.http.post(`${this.apiUrl}/reset`, data);
  }

  deleteAccount(): Observable<any> {
    return this.http.delete(`${this.apiUrl}/delete-account`, { withCredentials: true });
  }

  reactivateUser(data: { email: string, password: string }): Observable<any> {
    return this.http.post(`${this.apiUrl}/activate-user`, data);
  }

  getUserDetails(): Observable<any> {
    return this.http.get(`${this.apiUrl}/user-details`, { withCredentials: true });
  }

  isLoggedIn(): boolean {
    return !!localStorage.getItem('accessToken');
  }
}

